controllers/cartController.js

```diff
@@ -0,0 +1,47 @@
+import { carts, products } from '../storage.js';
+import { randomUUID } from 'crypto';
+
+export const addToCart = (req, res) => {
+  const { productId } = req.params;
+  const userId = req.headers['x-user-id'];
+
+  if (!userId) {
+    return res.status(401).json({ error: 'User ID is required' });
+  }
+
+  const product = products.find((p) => p.id === parseInt(productId, 10));
```

Comment:
I would suggest using more informative variable name, for example, `product` instead of `p`